nx-X11/programs/Xserver/os/connection.c

```diff
@@ -857,59 +859,44 @@ EstablishNewConnections(ClientPtr clientUnused, void * closure)
 		CloseDownClient(client);     
 	}
     }
-#ifndef WIN32
-    for (i = 0; i < howmany(XFD_SETSIZE, NFDBITS); i++)
-    {
-      while (readyconnections.fds_bits[i])
-#else
-      for (i = 0; i < XFD_SETCOUNT(&readyconnections); i++) 
-#endif
-      {
-	XtransConnInfo trans_conn, new_trans_conn;
-	int status;
-
-#ifndef WIN32
-	curconn = ffs (readyconnections.fds_bits[i]) - 1;
-	readyconnections.fds_bits[i] &= ~((fd_mask)1 << curconn);
-	curconn += (i * (sizeof(fd_mask)*8));
-#else
-	curconn = XFD_FD(&readyconnections, i);
-#endif
+    if ((trans_conn = lookup_trans_conn(curconn)) == NULL)
+	return TRUE;
 
-	if ((trans_conn = lookup_trans_conn (curconn)) == NULL)
-	    continue;
+    if ((new_trans_conn = _XSERVTransAccept(trans_conn, &status)) == NULL)
+	return TRUE;
 
-	if ((new_trans_conn = _XSERVTransAccept (trans_conn, &status)) == NULL)
-	    continue;
+    newconn = _XSERVTransGetConnectionNumber(new_trans_conn);
 
-	newconn = _XSERVTransGetConnectionNumber (new_trans_conn);
+    if (newconn < lastfdesc) {
+	int clientid;
 
-	if (newconn < lastfdesc)
-	{
-		int clientid;
 #if !defined(WIN32)
-  		clientid = ConnectionTranslation[newconn];
+	clientid = ConnectionTranslation[newconn];
 #else
-  		clientid = GetConnectionTranslation(newconn);
+	clientid = GetConnectionTranslation(newconn);
 #endif
-		if(clientid && (client = clients[clientid]))
- 			CloseDownClient(client);
-	}
+	if (clientid && (client = clients[clientid]))
+	    CloseDownClient(client);
+    }
 
-	_XSERVTransSetOption(new_trans_conn, TRANS_NONBLOCKING, 1);
+    _XSERVTransSetOption(new_trans_conn, TRANS_NONBLOCKING, 1);
+
+    if (trans_conn->flags & TRANS_NOXAUTH)
+        new_trans_conn->flags = new_trans_conn->flags | TRANS_NOXAUTH;
+
+    if (!AllocNewConnection(new_trans_conn, newconn, connect_time)) {
+	ErrorConnMax(new_trans_conn);
+     }
 
-	if (!AllocNewConnection (new_trans_conn, newconn, connect_time
-				))
-	{
-	    ErrorConnMax(new_trans_conn);
-	}
-      }
-#ifndef WIN32
-    }
-#endif
     return TRUE;
 }
 
+static void
```

Comment:
Something in this commit looks odd...

The upstream commit 7b02f0b87ec2fa0cc5a65307a1fd55c671cec884 has this hunk:

```diff
@@ -1417,8 +1398,7 @@ ListenOnOpenFD(int fd, int noxauth)
     ListenTransConns[ListenTransCount] = ciptr;
     ListenTransFds[ListenTransCount] = fd;

-    FD_SET(fd, &WellKnownConnections);
-    FD_SET(fd, &AllSockets);
+    SetNotifyFd(fd, QueueNewConnections, X_NOTIFY_READ, NULL);

     /* Increment the count */
     ListenTransCount++;
```

which is missing in your backport.

While the function in question doesn't exist in our source tree, I hope we're not missing out on the `SetNotifyFd()` call. Probably not, since the function is not even used on anything but the Darwin and Wayland DDX'.